packages/cosmos/src/client.rs

```diff
@@ -1289,6 +1349,14 @@ mod tests {
             None
         );
     }
+
+    #[test]
+    fn gas_estimate_multiplier() {
```

Comment:
@lvn-rusty-dragon heads up, I think it might help to review starting from the test, since this PR is more about the API than the internal workings (though ofc both are important)